hub/hub_service.bal

```diff
@@ -267,24 +261,44 @@ websubhub:Service hubService = service object {
     }
 
     isolated function updateMessage(websubhub:UpdateMessage msg) returns websubhub:UpdateMessageError? {
+       
         string topicName = util:sanitizeTopicName(msg.hubTopic);
+        self.createTopicIFNotExist(topicName);
+        log:printInfo("Running content update", topic = msg.hubTopic);
+        error? errorResponse = persist:addUpdateMessage(topicName, msg);
+        if errorResponse is websubhub:UpdateMessageError {
+            return errorResponse;
+        } else if errorResponse is error {
+            log:printError("Error occurred while publishing the content ", errorMessage = errorResponse.message(), topic = topicName);
+            return error websubhub:UpdateMessageError(errorResponse.message());
+        }
+
+    }
+
+    isolated function createTopicIFNotExist(string topicName) {
         boolean topicAvailable = false;
         lock {
             topicAvailable = registeredTopicsCache.hasKey(topicName);
         }
-        if topicAvailable {
-            log:printInfo("Running content update", topic = msg.hubTopic);
-            error? errorResponse = persist:addUpdateMessage(topicName, msg);
-            if errorResponse is websubhub:UpdateMessageError {
-                return errorResponse;
-            } else if errorResponse is error {
-                log:printError("Error occurred while publishing the content ", errorMessage = errorResponse.message(),topic=topicName);
-                return error websubhub:UpdateMessageError(errorResponse.message());
+
+        if !topicAvailable {
+            log:printInfo("Topic not found - Auto registering topic", topic = topicName);
+            websubhub:TopicRegistration topicRegistrationMsg = {
+                topic: topicName,
+                hubMode: "register"
+            };
+            websubhub:TopicRegistrationError? registerTopicResult = self.registerTopic(topicRegistrationMsg);
+            if registerTopicResult is websubhub:TopicRegistrationError {
+                log:printError("Auto registering topic - Successfully registered failed with error", err = registerTopicResult.message());
```

Comment:
In this case, whyt we are not returning any error to publish or subscribe API ?

Reply:
there is only one error returned by api registered topic already exist, we will not stop program for that right?